### PR TITLE
Add Device Binding documentation

### PR DESCRIPTION
**Pull request by dmytroyarmak**
Updated also docs under https://www.notion.so/kontist/2FA-via-Device-Binding-92ad2c3743a1409db6123e5e46716d63

Addresses https://app.zenhub.com/workspaces/kontist-engineering-5be06ab44b5806bc2bf14eca/issues/figureaps/figure-backend/5499

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
...
+To initiate the device binding you need to create a device. After a successful request, you will receive an SMS with OTP that you will need during device verification.
+
+```shell
+curl "https://api.kontist.com/api/user/devices" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
+  -H "Content-Type: application/json" \
+  -X POST \
+  -d '{
+        "name": "iPhone XS",
+        "key": "0402e86575939cd541f016b69b1bc6ee97736f7a6d32c0ad375695ffdc03acf21a3b54224fd164ad6f9cfdfb42b74f49f3d34a41f95d62e893be4977c7ec154f29"
```
Comment (johannespfeiffer) [inline source/index.html.md:132]: Let's add a few words 
* on the expected format (e.g. without the header, just "raw" seems a bit to ambiguous)
* emphasize that we always only expect the public key
* the private key should be stored in a secure place.

File: source/index.html.md [hunk 2]
```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
...
+
+After that, you can use your device id and your private key to get your confirmed auth token by creating and verifying device challenge.
+
+### Create device
+
+To initiate the device binding you need to create a device. After a successful request, you will receive an SMS with OTP that you will need during device verification.
+
+```shell
+curl "https://api.kontist.com/api/user/devices" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
```
Comment (johannespfeiffer) [inline source/index.html.md:127]: Do we need to use Basic Auth here? I think we should use our unconfirmed JWT

File: source/index.html.md [hunk 3]
```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
...
+| deviceId    | ID of the device                                                               |
+| challengeId | ID of the challenge                                                            |
+
+### Verify device
+
+To verify device you need to provide a signature of OTP received on your mobile phone.
+
+```shell
+curl "https://api.kontist.com/api/user/devices/4e310a55-1b1a-4efb-b9a5-fd04491bdd21/verify" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
```
Comment (johannespfeiffer) [inline source/index.html.md:168]: Do we need to use Basic Auth here? I think we should use our unconfirmed JWT

File: source/index.html.md [hunk 4]
```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
...
+| challengeId | yes       | ID of the challenge recieved during device creation                            |
+| signature   | yes       | The hex-encoded, raw signature for the OTP recived in SMS                      |
+
+### Create device challenge
+
+After the device is created and verified, you need to create a device challenge. As a response, you will receive `stringToSign` that should be used during verification of device challenge.
+
+```shell
+curl "https://api.kontist.com/api/user/devices/4e310a55-1b1a-4efb-b9a5-fd04491bdd21/challenges" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
```
Comment (johannespfeiffer) [inline source/index.html.md:196]: Do we need to use Basic Auth here? I think we should use our unconfirmed JWT

File: source/index.html.md [hunk 5]
```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
...
+| stringToSign  | Challenge string that should be signed by device private key                   |
+
+
+### Verify device challenge
+
+To verify device challenge you need to provide a signature of `stringToSign` received after challenge creation.
+
+```shell
+curl "https://api.kontist.com/api/user/devices/4e310a55-1b1a-4efb-b9a5-fd04491bdd21/challenges/5f7c36e2-e0bf-4755-8376-ac6d0711192e/verify" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
```
Comment (johannespfeiffer) [inline source/index.html.md:226]: Do we need to use Basic Auth here? I think we should use our unconfirmed JWT

File: source/index.html.md [hunk 6]
```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
...
+  -H "Content-Type: application/json" \
+  -X POST \
+  -d '{
+        "signature": "FF93DF062DB808E35AB9D28D80E0B261C7313C69785471954C05A474156BA7A7F07F2F0E7E805513754A8119BBF172E1E6D0103901249CE8DE012E5E61FDA36AD06405341043"
+      }'
+```
+
+> The above command returns JSON structured like this:
+
+```json
```
Comment (johannespfeiffer) [inline source/index.html.md:236]: Does it make sense to decode this token and give a hint how you can distinguish a confirmed from an unconfirmed token (cnf.kid)?
Or do we say it is an implementation detail?

**Review by johannespfeiffer — CHANGES_REQUESTED** [review]
Looks good, just minor comments/questions.

File: source/index.html.md (hunk 1 above)
Comment (dmytroyarmak) [reply source/index.html.md:132]: Thanks, I've added additional explanation

File: source/index.html.md (hunk 6 above)
Comment (dmytroyarmak) [reply source/index.html.md:236]: I think it's not needed for API users to know about it. Let's leave it as implementation detail :)

File: source/index.html.md [hunk 7]
```diff
@@ -104,6 +104,157 @@ curl -X DELETE \
...
+
+#### HTTP Request
+
+`POST https://api.kontist.com/api/user/devices/{device_id}/challenges/{challenge_id}/verify`
+
+#### Request body
+
+| Parameter   | Mandatory | Description                                                                    |
+| ----------- | --------- | ------------------------------------------------------------------------------ |
+| signature   | yes       | The hex-encoded signature for the `stingToSign`                                |
```
Comment (johannespfeiffer) [inline source/index.html.md]: minor typo, missing "r" in `stingToSign`

**Comment by dmytroyarmak** [comment]
@johannespfeiffer You are right that basic auth won't be used after we enforce 2FA, but let's leave it as it is for now. 
I would better change all request examples from `Basic` to `Bearer` in a separate pull request.

File: source/index.html.md (hunk 2 above)
Comment (dmytroyarmak) [reply source/index.html.md:127]: answered below

File: source/index.html.md (hunk 3 above)
Comment (dmytroyarmak) [reply source/index.html.md:168]: answered below

File: source/index.html.md (hunk 5 above)
Comment (dmytroyarmak) [reply source/index.html.md:226]: answered below

File: source/index.html.md (hunk 7 above)
Comment (dmytroyarmak) [reply source/index.html.md]: thanks, fixed

**Review by johannespfeiffer — APPROVED** [review]
Looks good!

File: source/index.html.md [hunk 8]
```diff
@@ -104,6 +104,157 @@ curl -X DELETE \
 
 You can access Kontist API endpoints by adding `Authorization`: `Bearer AUTH_TOKEN` header to your requests.
 
+## Device Binding
+
+To have access to Kontist API endpoints that require strong customer authentication, you need to pass two-factor authentication (2FA).
+
+To make it seamless for you we provide device binding authentication that leverages digital signature algorithm.
+
+To pass device binding authentication you need to generate a pair of private and public keys by using elliptic curve algorithm  (secp256r1). To create and verify your device you need to pass your public key and then sign OTP from SMS received on your phone.
```
Comment (hihuz) [inline source/index.html.md]: nit: there is an extra space between `algorithm` and `(secp256r1)`

**Review by hihuz — APPROVED** [review]
LGTM 👍

File: source/index.html.md (hunk 8 above)
Comment (dmytroyarmak) [reply source/index.html.md]: thanks, fixed

**Pull request closed without merging**